### PR TITLE
ctre: conan v2 support

### PR DESCRIPTION
**Pull request by StellaSmith**

### CODE REVIEW

**Comment by StellaSmith** [comment]
@SpaceIm not to be annoying but might I have some reviews?

File: recipes/ctre/all/test_v1_package/test_package.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#include <ctre.hpp>
```
Comment (SpaceIm) [inline recipes/ctre/all/test_v1_package/test_package.cpp:1]: not need to duplicate source code of test package. Please take a look at recipes templates in docs.

File: recipes/ctre/all/test_v1_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(ctre CONFIG REQUIRED)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ctre::ctre)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
```
Comment (SpaceIm) [inline recipes/ctre/all/test_v1_package/CMakeLists.txt:12]:
```suggestion
cmake_minimum_required(VERSION 3.1)
project(test_package)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package)
```

File: recipes/ctre/all/test_v1_package/test_package.cpp (hunk 1 above)
Comment (SpaceIm) [inline recipes/ctre/all/test_v1_package/test_package.cpp:1]: this file can be removed

File: recipes/ctre/all/conanfile.py [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
 from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+from conan.tools.build import check_min_cppstd
+from conan.tools.layout import basic_layout
+from conan.tools.files import get, copy
 
 required_conan_version = ">=1.33.0"
```
Comment (SpaceIm) [inline recipes/ctre/all/conanfile.py:10]:
```suggestion
required_conan_version = ">=1.50.0"
```

File: recipes/ctre/all/conanfile.py [hunk 4]
```diff
@@ -46,12 +51,11 @@ def validate(self):
...
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def package(self):
-        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-
+        copy(self, pattern="*.hpp", src=os.path.join(self.source_folder, "include"), dst=os.path.join(self.package_folder, "include"))
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```
Comment (SpaceIm) [inline recipes/ctre/all/conanfile.py:61]:
```suggestion
        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
        
        def package_info(self):
            self.cpp_info.bindirs = []
            self.cpp_info.libdirs = []
```

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
Looking great, sorry again for the delay caused by the holidays and the 2.0 launch!
lgtm

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**